mongodb-datastore/handlers/events.go

```diff
@@ -275,39 +295,46 @@ func dropProjectEvents(logger *keptncommon.Logger, event *models.KeptnContextExt
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 	defer cancel()
 
-	collectionName := getProjectOfEvent(event)
-	collection := client.Database(mongoDBName).Collection(collectionName)
-	rootEventsCollection := client.Database(mongoDBName).Collection(collectionName + rootEventCollectionSuffix)
-	invalidatedEventsCollection := client.Database(mongoDBName).Collection(getInvalidatedCollectionName(collectionName))
+	projectName := getProjectOfEvent(event)
+	projectCollection := client.Database(mongoDBName).Collection(projectName)
+	rootEventsCollection := client.Database(mongoDBName).Collection(projectName + rootEventCollectionSuffix)
+	invalidatedEventsCollection := client.Database(mongoDBName).Collection(getInvalidatedCollectionName(projectName))
+
+	logger.Debug(fmt.Sprintf("Delete all events of project %s", projectName))
 
-	logger.Debug(fmt.Sprintf("Delete all events of project %s", collectionName))
-	err := collection.Drop(ctx)
+	createdIndexes[projectCollection.Name()+"-shkeptncontext"] = false
+	createdIndexes[projectCollection.Name()+"-type"] = false
+	createdIndexes[projectCollection.Name()+"-data.service"] = false
```

Comment:
The code above, as well as the code a couple lines below
```
createdIndexes[rootEventsCollection.Name()+"-data.service"] = false
createdIndexes[rootEventsCollection.Name()+"-time"] = false
```

is painful to maintain - as in: We will forget about it in the future :)

If anything, at the end of `dropProjectEvents` we could have an assert or something to check if `createdIndexes` contains anything that starts with the project name (maybe the separator `-` can be replaced by something that's not allowed for project names?).

Reply:
good idea - using a separator that is not allowed within a keptn project name would do the trick. I will adjust this accordingly

Reply:
did it a little differently - I added arrays containing the names of each index to be created for each collection. When deleting the collections for a project, I iterate over them and make sure the correlating values in the `skipCreateIndex` map are set to `false`. Also, when creating the indexes the service will use those arrays, which makes the code cleaner